site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+converter is not able to convert the request.
+
+.. code-block:: java
+
+    class MyRequestConverter implements RequestConverterFunction {
+        @Override
+        public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage request,
+                                     Class<?> expectedResultType) throws Exception {
+            if (expectedResultType == MyObject.class) {
+                // Convert the request to a Java object.
+                return new MyObject(request.content());
+            }
+
+            // To the next request converter.
+            return RequestConverterFunction.fallthrough();
+        }
+    }
+
+Then, you can write your service method as follows. Note that a request converter will work on the parameters
+which are annotated with `@RequestObject`_.
+
+.. code-block:: java
+
+    @RequestConverter(MyRequestConverter.class)
+    class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public HttpResponse hello(@RequestObject MyObject myObject) {
+            // MyRequestConverter will be used to convert a request.
+            ...
+        }
+
+        @Post("/hola")
+        @RequestConverter(MySpanishRequestConverter.class)
+        public HttpResponse hola(@RequestObject MySpanishObject myObject) {
+            // MySpanishRequestConverter will be tried to convert a request first.
+            // MyRequestConverter will be used if MySpanishRequestConverter fell through.
+            ...
+        }
+    }
+
+Armeria also provides built-in request converters such as, JacksonRequestConverterFunction_ for JSON
+documents, StringRequestConverterFunction_ for text contents and ByteArrayRequestConverterFunction_
+for binary contents. They will be applied after your request converters by default, so you can use these
+built-in converters by just putting `@RequestObject`_ annotation on the parameters which you want to convert.
+
+In some cases, `@RequestObject`_ annotation may have a request converter as its value.
+Assume that you have a Java class named ``MyRequest`` that it is usually able to be converted by
+``MyDefaultRequestConverter``. But what if there is only one method which has a parameter of ``MyRequest``
+that you have to convert it differently. In this case, you may specify a request converter with
+`@RequestObject`_ annotation. In the example, ``MySpecialRequestConverter`` will be used first for
+converting ``MyRequest``.
+
+.. code-block:: java
+
+    @RequestConverter(MyDefaultRequestConverter.class)
+    class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public HttpResponse hello(
+            @RequestObject(MySpecialRequestConverter.class) MyRequest myRequest) { ... }
+    }
+
+Converting a Java object to an HTTP response
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Every object returning by the annotated service method is able to convert to an HTTP response message by
```

Comment:
- returning by .. -> returned by an annotated ...
- is able to convert to -> can be converted to